packages/snaps-controllers/src/snaps/location/npm.ts

```diff
@@ -267,6 +270,13 @@ async function fetchNpmTarball(
   if (!tarballResponse.ok || !tarballResponse.body) {
     throw new Error(`Failed to fetch tarball for package "${packageName}".`);
   }
+  const tarballSizeString = tarballResponse.headers.get('content-length');
+  assert(tarballSizeString, 'Snap tarball has invalid content-length');
+  const tarballSize = parseInt(tarballSizeString, 10);
+  assert(
+    tarballSize <= TARBALL_SIZE_SAFETY_LIMIT / 2,
```

Comment:
Why the division by 2?

Reply:
Assuming compression of 50% or so it seemed fine to reduce the limit a bit since the tarball should be smaller than the unpacked files 😅